AEPIdentity/Sources/Identity.swift

```diff
@@ -180,7 +184,14 @@ import Foundation
     private func processAppendToUrl(baseUrl: String, event: Event) {
         guard let properties = state?.identityProperties else { return }
         guard let configurationSharedState = getSharedState(extensionName: IdentityConstants.SharedStateKeys.CONFIGURATION, event: event)?.value else { return }
-        let analyticsSharedState = getSharedState(extensionName: "com.adobe.module.analytics", event: event)?.value ?? [:]
+
+        let analyticsRegistered = isExtensionRegistered(name: IdentityConstants.SharedStateKeys.ANALYTICS, event: event)
+
+        if analyticsRegistered && getSharedState(extensionName: IdentityConstants.SharedStateKeys.ANALYTICS, event: event)?.status == .pending {
+            return
```

Comment:
I think all this work needs to be done in `readyForEvent`. Returning from the function simply stops processing the event, but it won't keep it in the queue. Additionally, without a dispatched response, the public API will simply time out.